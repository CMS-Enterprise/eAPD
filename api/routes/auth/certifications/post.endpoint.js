import {
  getDB,
  setupDB,
  teardownDB,
  apiAsFedAdmin,
  unauthenticatedTest,
  unauthorizedTest
} from '../../../endpoint-tests/utils.js';

describe('auth/certifications endpoints', () => {
  const db = getDB();
  const api = apiAsFedAdmin;
  beforeAll(async () => {
    await setupDB(db);
  });
  afterAll(async () => {
    await teardownDB(db);
  });

  describe('POST /auth/certifications', () => {
    const validRequestBody = {
      ffy: 2021,
      name: 'Roger Klotz',
      email: 'fake@email.com',
      state: 'na',
      fileUrl: '/auth/certifications/files/12345'
    };

    const url = '/auth/certifications';

    unauthenticatedTest('post', url);
    unauthorizedTest('post', url);

    describe('when authenticated as a user with permission', () => {
      it('with no request body', async () => {
        const response = await api.post(url, {});

        expect(response.status).toEqual(400);
        expect(response.data).toMatchSnapshot();
      });

      it('with a valid request body', async () => {
        const response = await api.post(url, validRequestBody);

        expect(response.status).toEqual(200);
      });
    });
  });
});
