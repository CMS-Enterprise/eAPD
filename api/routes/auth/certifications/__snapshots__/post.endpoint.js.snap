// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth/certifications endpoints POST /auth/certifications when authenticated as a user with permission with no request body 1`] = `
Object {
  "error": "Unable to save state admin certification",
}
`;
